io.go

```diff
@@ -360,33 +364,261 @@ func (c *Cell) setEmissionsFlux(e *Emissions) {
 	}
 }
 
-// Output returns a function that writes simulation results to a shapefile or
-// shapefiles.
-// If  allLayers` is true, the function writes out data for all of the vertical
-// layers, otherwise only the ground-level layer is written.
-// outputVariables is a map of the names of the variables for which data
-// should be returned to expressions that define how the data should be calculated.
-// These expressions can contain built-in InMAP variables, user-defined variables,
-// and functions. For more information on the functions available for defining
-// output variables see the documentation for the Results function.
-func Output(fileName string, allLayers bool, outputVariables map[string]string) DomainManipulator {
+// Outputter is a holder for output parameters.
+type Outputter struct {
+	fileName        string
+	AllLayers       bool
+	OutputVariables map[string]string
```

Comment:
Is there a reason OutputVariables, GetVariables, and OutputFunctions need to be exported here. Also, perhaps rename GetVariables to ModelVariables or ModelVars or something like that.